owncloudApp/src/main/java/com/owncloud/android/shares/ui/fragment/ShareFileFragment.kt

```diff
@@ -489,6 +524,16 @@ class ShareFileFragment : Fragment(), ShareUserListAdapter.ShareUserAdapterListe
             sharePublicLinksList?.visibility = View.GONE
         }
 
+        // Show or hide button for adding a new public share depending on the capabilities and
+        // the server version
+        if (!enableMultiplePublicSharing()) {
+            if (publicLinks?.size == 0) {
+                addPublicLinkButton.visibility = View.VISIBLE;
+            } else if (publicLinks?.size!! >= 1) {
+                addPublicLinkButton.visibility = View.INVISIBLE;
```

Comment:
Why do you need to use (`publicLinks?.size!!`) here but not in the previous if (`publicLinks?.size == 0`)?

Reply:
`==` calls `equals(other: Any?): Boolean` while `>` calls `compareTo(other: Int): Int`